// react-messenger-checkbox/lib/MessengerCheckbox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var MessengerCheckbox = function (_Component) {
  _inherits(MessengerCheckbox, _Component);

  function MessengerCheckbox() {
    _classCallCheck(this, MessengerCheckbox);

    return _possibleConstructorReturn(this, (MessengerCheckbox.__proto__ || Object.getPrototypeOf(MessengerCheckbox)).apply(this, arguments));
  }

  _createClass(MessengerCheckbox, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      if (document.getElementById('facebook-jssdk')) {
        return;
      }
      this.setFbAsyncInit();
      this.loadSdkAsynchronously();
    }
  }, {
    key: 'setFbAsyncInit',
    value: function setFbAsyncInit() {
      var _props = this.props,
          appId = _props.appId,
          pageId = _props.pageId,
          autoLogAppEvents = _props.autoLogAppEvents,
          xfbml = _props.xfbml,
          version = _props.version,
          onEvent = _props.onEvent,
          userRef = _props.userRef;

      window.fbAsyncInit = function () {
        window.FB.init({
          appId: appId,
          autoLogAppEvents: autoLogAppEvents,
          xfbml: xfbml,
          version: 'v' + version
        });

        window.FB.Event.subscribe('messenger_checkbox', onEvent);

        window.confirmOptIn = function confirmOptIn(ref) {
          window.FB.AppEvents.logEvent('MessengerCheckboxUserConfirmation', null, {
            app_id: appId,
            page_id: pageId,
            ref: ref,
            user_ref: userRef
          });
        };
      };
    }
  }, {
    key: 'loadSdkAsynchronously',
    value: function loadSdkAsynchronously() {
      var _props2 = this.props,
          language = _props2.language,
          debug = _props2.debug;
      /* eslint-disable */

      (function (d, s, id) {
        var js,
            fjs = d.getElementsByTagName(s)[0];
        if (d.getElementById(id)) {
          return;
        }
        js = d.createElement(s);
        js.id = id;
        js.src = 'https://connect.facebook.net/' + language + '/sdk' + (debug ? '/debug' : '') + '.js';
        fjs.parentNode.insertBefore(js, fjs);
      })(document, 'script', 'facebook-jssdk');
      /* eslint-enable */
    }
  }, {
    key: 'createMarkup',
    value: function createMarkup() {
      var _props3 = this.props,
          appId = _props3.appId,
          pageId = _props3.pageId,
          origin = _props3.origin,
          userRef = _props3.userRef,
          prechecked = _props3.prechecked,
          allowLogin = _props3.allowLogin,
          size = _props3.size,
          skin = _props3.skin,
          centerAlign = _props3.centerAlign;


      return {
        __html: '<div class="fb-messenger-checkbox"\n        origin="' + origin + '"\n        page_id="' + pageId + '"\n        messenger_app_id="' + appId + '"\n        user_ref="' + userRef + '"\n        prechecked="' + prechecked + '"\n        allow_login="' + allowLogin + '"\n        size="' + size + '"\n        skin="' + skin + '"\n        center_align="' + centerAlign + '"\n      ></div>'
      };
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement('div', { dangerouslySetInnerHTML: this.createMarkup() });
    }
  }]);

  return MessengerCheckbox;
}(_react.Component);

MessengerCheckbox.propTypes = {
  pageId: _propTypes2.default.string.isRequired,
  appId: _propTypes2.default.string.isRequired,
  origin: _propTypes2.default.string.isRequired,
  userRef: _propTypes2.default.string.isRequired,

  prechecked: _propTypes2.default.bool,
  allowLogin: _propTypes2.default.bool,
  size: _propTypes2.default.oneOf(['small', 'medium', 'large', 'standard', 'xlarge']),
  skin: _propTypes2.default.oneOf(['light', 'dark']),
  centerAlign: _propTypes2.default.bool,
  autoLogAppEvents: _propTypes2.default.bool,
  xfbml: _propTypes2.default.bool,
  version: _propTypes2.default.string,
  language: _propTypes2.default.string,
  debug: _propTypes2.default.bool,
  onEvent: _propTypes2.default.func
};
MessengerCheckbox.defaultProps = {
  prechecked: true,
  allowLogin: true,
  size: 'large',
  skin: 'light',
  centerAlign: false,
  autoLogAppEvents: true,
  xfbml: true,
  version: '2.11',
  language: 'en_US',
  debug: false,
  onEvent: function onEvent() {}
};
exports.default = MessengerCheckbox;
